### ISSUE TITLE
Swift 4 SWIFT_CLASS produces symbol duplication (Swift 4)

### ISSUE THREAD

**Issue by zulkis** [Quick/Nimble#477]
- Nimble: *7.0.2*
 - Xcode Version: *9.1 beta (9B37)*
 - Swift Version: *4*
 - Platform: *MacOS*

I am using cocoapods with next structure:
```
use_frameworks!

$rxSwiftVersion = '~> 4.0'
$resultVersion = '~> 3.0'

...

target :'MyKit-macOS' do
    platform :osx, '10.13'

    pod 'RxSwift', $rxSwiftVersion
    pod 'Result', $resultVersion
end

target :'MyKit-macOSTests' do
    platform :osx, '10.13'

    pod 'RxBlocking', $rxSwiftVersion
    pod 'RxTest', $rxSwiftVersion
    pod 'Nimble'
end
```

### What did you do?

build MyKit for testing. Nimble framework could not be built because of duplicated symbols occurring in DSL.m "SWIFT_CLASS("_TtC6Nimble7NMBWait") ..." and CurrentTestCaseTracker.h
"SWIFT_CLASS("_TtC6Nimble22CurrentTestCaseTracker")"

### What did you expect to happen?

I expect nimble to be built normally ;)

If I remove this SWIFT_CLASS declaration of interfaces in ObjC files - then it is building normally.
I expect that currently with Swift 4 Objective-C could get symbols marked with @objc without any issues, and therefore it is conflicting during compilation time with this particular declarations.

After removing SWIFT_CLASS declaration I tried unit tests for both ObjC and Swift:

```
@interface ObjCTest : XCTestCase

@end

@implementation ObjCTest
- (void)testExample {
    [NMBWait untilTimeout:3 file:@"" line:1 action:^(void (^ _Nonnull callback)(void)) {
        callback();
    }];
}
@end
```

```
import Nimble

class ObjcTest: XCTestCase {
    func testSwift() {
        waitUntil { (callback) in
            callback()
        }
    }
}
```

Works perfectly and getting references to swift side of this methods.

**Comment by Constantine-Fry** [comment]
The solution is to set `APPLICATION_EXTENSION_API_ONLY` to `NO` on project level, but keep on for targets.

https://github.com/Quick/Quick/issues/749#issuecomment-343232565

**Comment by ikesyo** [comment]
Addressed by #525.
